components/raftstore-v2/src/router/message.rs

```diff
@@ -0,0 +1,293 @@
+            ResponseChannel::Read(read) => {
+                read.notify_read(Err(resp));
+            }
+            ResponseChannel::Write(write) => {
+                write.notify_applied(Err(resp));
+            }
+        }
+    }
+
+    pub fn notify_proposed(&mut self) {
+        if let ResponseChannel::Write(write) = self {
+            write.notify_proposed();
+        }
+    }
+
+    pub fn notify_committed(&mut self) {
+        if let ResponseChannel::Write(write) = self {
+            write.notify_committed();
+        }
+    }
+
+    pub fn invoke_read(self, args: ReadResponse<S>) {
+        match self {
+            ResponseChannel::Read(read) => read.notify_read(Ok(args)),
+            other => panic!("expect Callback::Read(..), got {:?}", other),
+        }
+    }
+
+    pub fn is_none(&self) -> bool {
+        matches!(self, ResponseChannel::None)
+    }
+}
+
+impl<S> fmt::Debug for ResponseChannel<S>
+where
+    S: Snapshot,
+{
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ResponseChannel::None => write!(fmt, "Callback::None"),
+            ResponseChannel::Read(_) => write!(fmt, "Callback::Read(..)"),
+            ResponseChannel::Write { .. } => write!(fmt, "Callback::Write(..)"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(u8)]
+pub enum PeerTick {
+    Raft = 0,
+    RaftLogGc = 1,
+    SplitRegionCheck = 2,
+    PdHeartbeat = 3,
+    CheckMerge = 4,
+    CheckPeerStaleState = 5,
+    EntryCacheEvict = 6,
+    CheckLeaderLease = 7,
+    ReactivateMemoryLock = 8,
+    ReportBuckets = 9,
+}
+
+impl PeerTick {
+    pub const VARIANT_COUNT: usize = Self::get_all_ticks().len();
+
+    #[inline]
+    pub fn tag(self) -> &'static str {
+        match self {
+            PeerTick::Raft => "raft",
+            PeerTick::RaftLogGc => "raft_log_gc",
+            PeerTick::SplitRegionCheck => "split_region_check",
+            PeerTick::PdHeartbeat => "pd_heartbeat",
+            PeerTick::CheckMerge => "check_merge",
+            PeerTick::CheckPeerStaleState => "check_peer_stale_state",
+            PeerTick::EntryCacheEvict => "entry_cache_evict",
+            PeerTick::CheckLeaderLease => "check_leader_lease",
+            PeerTick::ReactivateMemoryLock => "reactivate_memory_lock",
+            PeerTick::ReportBuckets => "report_buckets",
+        }
+    }
+
+    pub const fn get_all_ticks() -> &'static [PeerTick] {
+        const TICKS: &[PeerTick] = &[
+            PeerTick::Raft,
+            PeerTick::RaftLogGc,
+            PeerTick::SplitRegionCheck,
+            PeerTick::PdHeartbeat,
+            PeerTick::CheckMerge,
+            PeerTick::CheckPeerStaleState,
+            PeerTick::EntryCacheEvict,
+            PeerTick::CheckLeaderLease,
+            PeerTick::ReactivateMemoryLock,
+            PeerTick::ReportBuckets,
+        ];
+        TICKS
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum StoreTick {
```

Comment:
Same here. It seems identical to v1.

Reply:
No, they are different. Several ticks are removed.